bpf/process/generic_calls.h

```diff
@@ -652,8 +652,7 @@ FUNC_INLINE int generic_process_filter(void)
 	sel = &msg->sel;
 	current = &msg->current;
 
-	curr = sel->curr;
-	if (curr > MAX_SELECTORS)
+	if (sel->curr > MAX_SELECTORS)
```

Comment:
I'm surprised the compiler hasn't done this optimisation already?

Reply:
Yeah...

One possible explanation is that an optimization for bpf is not an optimization for x86, and it might be challenging to balance those if the optimizations happen before the target-specific backend. BPF probably takes less precendence over x86 or other hw archs.